### PR TITLE
Add backdated post indicator

### PR DESCRIPTION
**Pull request by mozzius**
Backdated post indicator. Requires backend changes to actually work
<table>
  <thead>
    <tr>
      <th>Indicator</th>
      <th>Info</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>Screenshot 2024-11-11 at 20 58 24</td>
      <td>Screenshot 2024-11-11 at 20 46 54</td>
    </tr>
  </tbody>
</table>

<table>
  <thead>
    <tr>
      <th>Indicator</th>
      <th>Info</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>Screenshot 2024-11-11 at 20 58 41</td>
      <td>Screenshot 2024-11-11 at 20 58 47</td>
    </tr>
  </tbody>
</table>

# Test plan

Replace `post.indexedAt` with the current date (or go back to before the final commit)

```diff
-const indexedAt = new Date(post.indexedAt)
+const indexedAt = new Date()
```

Confirm [posts older than 24 hours](https://bsky.app/profile/durial321.bsky.social/post/3l7bq5kibp42s) show the indicator

Confirm that the wording is good

### CODE REVIEW

File: src/view/com/post-thread/PostThreadItem.tsx [hunk 1]
```diff
@@ -758,6 +760,84 @@ function ExpandedPostDetails({
...
+      )}
+      <NewText style={[a.text_sm, t.atoms.text_contrast_medium]}>
+        {niceDate(i18n, post.indexedAt)}
+      </NewText>
+
+      <Prompt.Basic
+        control={control}
+        title={_(msg`Archival post`)}
+        description={_(
+          msg`This post claims to have been created at ${niceDate(
```
Comment (surfdude29) [inline src/view/com/post-thread/PostThreadItem.tsx:828]:
```suggestion
          msg`This post claims to have been created on ${niceDate(
```
wording is great imho, but I think using `on` would flow better and avoid repeating `at`

**Review by haileyok — APPROVED** [review]
code lgtm but have not tested (dont really know how to test, can test if theres a way to!)

**Comment by surfdude29** [comment]
> code lgtm but have not tested (dont really know how to test, can test if theres a way to!)

Sorry for jumping in here, this looks like it might be a good way to test? :)

"**Test plan**

Replace post.indexedAt with the current date (or go back to before the final commit)"

**Comment by Lucifer-is-my-pet** [comment]
Hello.

Is there a discussion of this feature anywhere? I'm one of the many people trying to get settled in Bluesky using import for my tweets (it is hard moving to a new place with nothing). But with this I will make thousands of posts in one day and flood other people's feeds because the `created_at` of tweets is ignored. And I just want my stuff to be on my new account here.

The main idea with labelling imported posts is great and I get the problem with misleading dates. But why not switch date on the label and post date and indicate the "unreliable" date for all to see (as described here #6707)?

Or will it be better to just make a pull request with necessary changes and wait for a review? I am willing to try if it will get us closer to a satisfactory solution.

**Comment by jamesleesaunders** [comment]
> Hello.
> 
> Is there a discussion of this feature anywhere? I'm one of the many people trying to get settled in Bluesky using import for my tweets (it is hard moving to a new place with nothing).

Hi @Lucifer-is-my-pet you may be interested to know I have put a proposal pull request in for a half-way solution https://github.com/bluesky-social/social-app/pull/7466 this keeps parts of the 'Archived Post' messaging whilst restoring a timeline for imported posts.

**Pull request merged**